test/e2e.sh

```diff
@@ -0,0 +1,95 @@
+    oc debug node/${worker_node} -- cat /host/etc/os-release  > ${tmpdir}/etc-os-version
+
+    eval $(cat ${tmpdir}/etc-os-version | grep OSTREE_VERSION)
+
+    rhcos_version=${OSTREE_VERSION}
+    echo "INFO: Node RHCOS version: ${rhcos_version}"
+}
+
+# Check if driver-toolkit imagestream is available
+test_imagestream() {
+    oc get imagestream/driver-toolkit -n openshift
+    exit_on_error $? "imagestream/driver-toolkit not found"
+}
+
+# Check for latest and RHCOS tags
+test_imagestream_tags() {
+    echo "INFO: Checking imagestream/driver-toolkit tags:"
+    oc get imagestream/driver-toolkit -n openshift -o json | jq ".spec.tags[].name"
+    
+    image_latest=$(oc get imagestream/driver-toolkit -n openshift -o json \
+    	| jq ".spec.tags[] | select(.name == \"latest\") | .from.name")
+    echo "INFO: Image from latest tag: ${image_latest}"
+        
+    image_rhcos_tag=$(oc get imagestream/driver-toolkit -n openshift -o json \
+    	| jq ".spec.tags[] | select(.name == \"${rhcos_version}\") | .from.name")
+    echo "INFO: Image from RHCOS tag: ${image_rhcos_tag}"
+
+    [[ $image_latest != "" ]]
+    exit_on_error $? "imagestream/driver-toolkit latest tag has no image"
+
+    [[ $image_rhcos_tag != "" ]]
+    exit_on_error $? "imagestream/driver-toolkit RHCOS tag has no image"
+    
+    [[ $image_rhcos_tag == $image_latest ]]
+    exit_on_error $? "imagestream/driver-toolkit ${rhcos_version} tag does not match latest tag"
+}
+
+test_kernel_version() {
+    # Check that DTK contains the right kernel version
+    dtk_release=$(oc debug -f manifests/01-rhcos-version.yaml \
+                  --quiet \
+                  -n ${test_ns}\
+                  --image=${dtk_registry_image} \
+                  -- cat /etc/driver-toolkit-release.json)
+    dtk_kernel=$(echo "${dtk_release}" | jq -r .KERNEL_VERSION)
+
+    echo "INFO: Node kernel: ${node_kernel}"
+    echo "INFO: driver-toolkit kernel: ${dtk_kernel}"
+    
+    [[ ${dtk_kernel} ==  ${node_kernel} ]]
+    exit_on_error $? "Mismatch between node kernel and driver-toolkit kernel."
+}
+
+list_kernel_packages() {
+    echo "INFO: dnf list installed | grep kernel:"
+    # Check that DTK contains all the packages
+    oc debug -f <(echo "$MINI_POD_SPEC") \
+                     --quiet \
+                     -n ${test_ns}\
+                     --image=${dtk_registry_image} \
+                     -- \
+                     dnf list installed | grep kernel
+}
+
+exit_on_error() {
+    exit_code=$1
+    message=$2
+    if [ $exit_code -ne 0 ]; then
+        >&2 echo "ERROR: ${message}, exit code ${exit_code}."
+        exit $exit_code
+    fi
+}
+
+# Globals
+test_ns="driver-toolkit-test"
+tmpdir=$(mktemp -d)
+dtk_registry_image="image-registry.openshift-image-registry.svc:5000/openshift/driver-toolkit:latest"
+node_kernel=$(oc get nodes -ojsonpath={.items[].status.nodeInfo.kernelVersion})
```

Comment:
We probably need to quote here as well?
```suggestion
node_kernel=$(oc get nodes -ojsonpath='{.items[].status.nodeInfo.kernelVersion}')
```

Reply:
Added for consistency.